### PR TITLE
Migrate to GitHub Actions reusable workflows (1)

### PR DESCRIPTION
**Pull request by rnro**
### Motivation:

Migrate to GitHub Actions reusable workflows whilst keeping CI green.

### Modifications:

As the first step in migrating to use GitHub actions this PR provides the workflows which will be used in the future, at first in disabled state so that we can keep CI green during the changeover.

### Result:

Once this PR is merged we can:
* Enable the new CI
* In a future PR neutralize the old CI scripts
* Disable the old CI via GitHub
* In a future PR Delete the old CI scripts

### CODE REVIEW

**Review by simonjbeaumont — COMMENTED** [review]
Looks good, but this is missing the scheduled.yaml.

File: .github/workflows/scheduled.yml
Comment (simonjbeaumont) [inline .github/workflows/scheduled.yml:1]: Any reason this file doesn't have the C++ interop pipeline. I expected the scheduled workflow to have everything that PR does, except soundness.

File: .github/workflows/pull_request.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+name: PR
+
+on:
+    pull_request:
+      types: [opened, reopened, synchronize]
+
+jobs:
+    soundness:
+        name: Soundness
+        uses: apple/swift-nio/.github/workflows/soundness.yml@main
```
Comment (FranzBusch) [inline .github/workflows/pull_request.yml:10]:
```suggestion
        uses: swiftlang/github-workflows/.github/workflows/soundness.yml@main
```

File: .github/workflows/pull_request.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+            license_header_check_project_name: "SwiftASN1"
+            shell_check_enabled: false
+            unacceptable_language_check_enabled: false
+    unit-tests:
+        name: Unit tests
+        uses: apple/swift-nio/.github/workflows/unit_tests.yml@main
+        with:
+            linux_5_8_enabled: false
+            linux_5_9_arguments_override: "--explicit-target-dependency-import-check error"
+            linux_5_10_arguments_override: "--explicit-target-dependency-import-check error"
```
Comment (FranzBusch) [inline .github/workflows/pull_request.yml:28]: We are we not passing `--warnings-as-errros` here?

File: .github/workflows/scheduled.yml
Comment (FranzBusch) [reply .github/workflows/scheduled.yml:1]: I think we discussed this a few weeks back offline. We said that we only want to run a subset of the PR checks nightly. That also included the C++ interop. In the NIO repo we also only run that on PRs.

**Review by FranzBusch — CHANGES_REQUESTED** [review]

File: .github/workflows/pull_request.yml (hunk 2 above)
Comment (rnro) [reply .github/workflows/pull_request.yml:28]: Good catch, that should be there

File: .github/workflows/scheduled.yml
Comment (simonjbeaumont) [reply .github/workflows/scheduled.yml:1]: I recall us saying we wouldn't run soundness on the scheduled job. Maybe I wasn't in that discussion :)

**Review by FranzBusch — APPROVED** [review]

**Pull request merged**